### PR TITLE
Add JSON serialization support

### PR DESCRIPTION
**Pull request by mccoyp**
cc: @pakrym 

Added setters to type properties and default constructors to allow JSON serializing with System.Text.Json.

### CODE REVIEW

File: src/dotnet/APIView/APIView/FieldAPIV.cs [hunk 1]
```diff
@@ -10,16 +10,19 @@ namespace APIView
...
-        public bool IsStatic { get; }
-        public bool IsVolatile { get; }
+        public bool IsConstant { get; set; }
+        public bool IsReadOnly { get; set; }
+        public bool IsStatic { get; set; }
+        public bool IsVolatile { get; set; }
 
-        public object Value { get; }
+        public object Value { get; set; }
+
```
Comment (pakrym) [inline src/dotnet/APIView/APIView/FieldAPIV.cs]: nit: Avoid double newlines everywhere

File: src/dotnet/APIView/APIViewWeb/Models/AssemblyModel.cs [hunk 2]
```diff
@@ -47,11 +46,11 @@ public AssemblyModel(Stream stream, string fileName)
             AssemblyAPIV assembly = AssemblyAPIV.AssemblyFromStream(stream);
             if (assembly == null)
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Models/AssemblyModel.cs]: Does this ever happen anymore?

File: src/dotnet/APIView/APIView/TreeRendererAPIV.cs [hunk 3]
```diff
@@ -67,7 +64,7 @@ internal static void RenderText(FieldAPIV f, StringBuilder builder, int indents
         internal static void RenderText(MethodAPIV m, StringBuilder builder, int indents = 0)
         {
             AppendIndentsText(builder, indents);
-            if (!m.Attributes.IsEmpty)
+            if (m.Attributes.Length != 0)
```
Comment (pakrym) [inline src/dotnet/APIView/APIView/TreeRendererAPIV.cs]: `.Any()`

File: src/dotnet/APIView/APIViewWeb/Models/AssemblyModel.cs [hunk 4]
```diff
@@ -16,14 +16,13 @@ public AssemblyModel(IConfiguration configuration)
 
         public int Id { get; set; }
 
-        [Display(Name = "Display String")]
-        public string DisplayString { get; set; }
+        public string JsonSerialization { get; set; }
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Models/AssemblyModel.cs]: Why is JsonSerialized value on the model and not the `AssemblyAPIV` itself?

File: src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs [hunk 5]
```diff
@@ -16,7 +18,9 @@ public ReviewModel(BlobAssemblyRepository assemblyRepository)
 
         public async Task OnGetAsync(string id)
         {
-            AssemblyModel = await assemblyRepository.ReadAssemblyContentAsync(id);
+            string json = await assemblyRepository.ReadAssemblyContentAsync(id);
+            AssemblyAPIV assembly = JsonSerializer.Parse<AssemblyAPIV>(json);
```
Comment (pakrym) [inline src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Review.cshtml.cs]: Move serialization/deserialization code into the repository implementation as it's an implementation detail of how the model is stored.

File: src/dotnet/APIView/APIViewWeb/Models/AssemblyModel.cs (hunk 2 above)
Comment (mccoyp) [reply src/dotnet/APIView/APIViewWeb/Models/AssemblyModel.cs]: Initializing from a stream does - using a filepath is outdated now, though. Updated!

Comment (pakrym) [reply src/dotnet/APIView/APIViewWeb/Models/AssemblyModel.cs]: > Initializing from a stream does 

In what case?

**Pull request merged**